### ISSUE TITLE
Replacing not working on all data of tabulator table

### ISSUE THREAD

**Issue by reshma-tamang1** [olifolkerd/tabulator#1800]
Hello Oli,

Loving its features. 
But I have a small question.
I am new to Tabulator and I'm trying to use it in my project. The main implementation is that whenever I click in the link present in the table, it should remove every link formatting and replace it with the data only. But only 25-27 data gets replaced leaving other.It occurs when I give table height. However I need both table height, pagination and that implementation. 

I have attached the [jsfiddle](https://jsfiddle.net/96xgd0hw/2/) link too. I'm using tabulator v4.1.
Is it my  virtual dom issue or  am I missing any built in function that will help me to get the desired result??

**Comment by Alarich** [comment]
TL;DR:
- [https://jsfiddle.net/g62mr7c8/](JSFiddle) 
- Add variable "flag" to track if table element has been clicked already
- Formatter listens to global "flag" variable to output appropriate html
- New data field "Selected" to hold the data on which element the user clicked on
- Set default table index to "sn"
- Send SN field to funct and update the single row data to "Selected" = true on click
- Force redraw of everything by calling table.redraw(true)


The main issue your code is having is that nothing has changed the formatter for the columns after the click, thus when they are added to the table, they will still have the old design.
More importantly, using jQuery (or even vanilla JS) to modify any of these elements is a bad approach with this library, as you will only change the elements that are visible in dom. These things should be done through the Tabulator functions.

I've created a [https://jsfiddle.net/g62mr7c8/](JSFiddle) with a rough code of what you might need.

I've added some things there that might not occur to you at first, namely I've tied the red box with a data field "Selected". 
Also I've used the "sn" field to update the data on that row only. This requires the index field to be set to "sn" for the table, as it is "id" by default, which your data does not have.
The last change is to add the true parameter to table.redraw, as it forces the data to be redrawn as well, instead of only the heights/sizes of cells, which it does by default.

This could be improved further surely. I am not aware of ways to attach general options to the table so that there wouldn't be a need for a global flag, but perhaps I've missed something. 
In theory you could attach an additional parameter to the data to track if anything has been clicked already, but that seems wasteful on its' own.
You could also add the "Selected" variable dynamically on click to all rows and have the formatter simply check if "Selected" exists, if not, it's a href, if yes, it's either the red box and value or just value.

Anyway, hopefully it will lead you in the right direction.

**Comment by olifolkerd** [comment]
Hey @reshma-tamang1 

@Alarich is correct. you are facing the issue because you are trying to update the contents of the table manually.

Because Tabulator uses a virtual DOM only the visible rows actually exist, the other rows are created and destroyed in real time as you scroll.

The only safe way to change the display of the table is through the formatters.

In your case i would suggest that you remove the link column and replace it with a new column with a different formatter in when a user clicks the cell. You can use the ***deleteColumn*** and ***addColumn*** functions to achieve this.

Another approach would be to use your own custom formatter that references a variable outside the table and then call the ***redraw*** function on the table when it changes to refresh all the formatters.

Could i ask that in future you ask questions on Stack Overflow, this issues list is for Bug Reports and Feature Requests only.

Cheers

Oli :)
